test/controllers/update/update.confirmation.controller.spec.ts

```diff
@@ -0,0 +1,83 @@
+jest.mock("ioredis");
+jest.mock('../../../src/middleware/authentication.middleware');
+jest.mock("../../../src/utils/logger");
+jest.mock('../../../src/utils/application.data');
+jest.mock('../../../src/middleware/service.availability.middleware');
+jest.mock('../../../src/utils/session');
+
+import { authentication } from "../../../src/middleware/authentication.middleware";
+import request from "supertest";
+import { NextFunction, Request, Response } from "express";
+import app from "../../../src/app";
+import { UPDATE_CONFIRMATION_URL, UPDATE_PAYMENT_FEE } from "../../../src/config";
+import { logger } from "../../../src/utils/logger";
+import { serviceAvailabilityMiddleware } from "../../../src/middleware/service.availability.middleware";
+
+import {
+  ANY_MESSAGE_ERROR,
+  SERVICE_UNAVAILABLE,
+  UPDATE_CONFIRMATION_PAGE_TITLE,
+  UPDATE_CONFIRMATION_PAGE_REFERENCE_NUMBER
+} from "../../__mocks__/text.mock";
+import { deleteApplicationData, getApplicationData } from "../../../src/utils/application.data";
+import { APPLICATION_DATA_MOCK, ENTITY_OBJECT_MOCK, getSessionRequestWithExtraData, userMail } from "../../__mocks__/session.mock";
+import { get } from "../../../src/controllers/confirmation.controller";
+import { getLoggedInUserEmail } from "../../../src/utils/session";
+
+const req = {} as Request;
+const res = { render: jest.fn() as any } as Response;
+
+const mockLoggerDebugRequest = logger.debugRequest as jest.Mock;
+const mockAuthenticationMiddleware = authentication as jest.Mock;
+mockAuthenticationMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => next() );
+const mockServiceAvailabilityMiddleware = serviceAvailabilityMiddleware as jest.Mock;
+mockServiceAvailabilityMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => next() );
+
+const mockGetApplicationData = getApplicationData as jest.Mock;
+const mockDeleteApplicationData = deleteApplicationData as jest.Mock;
+const mockGetLoggedInUserEmail = getLoggedInUserEmail as jest.Mock;
+
+describe("UPDATE CONFIRMATION controller", () => {
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockGetLoggedInUserEmail.mockReturnValue(userMail);
+  });
+
+  test("renders the update confirmation page", async () => {
+    mockGetApplicationData.mockReturnValueOnce(
+      { ...APPLICATION_DATA_MOCK }
+    );
+
+    const resp = await request(app).get(UPDATE_CONFIRMATION_URL);
+    expect(resp.status).toEqual(200);
```

Comment:
Test fails on line 53 - response returned is 500

Reply:
Fix pushed